tools/cldr-apps/src/org/unicode/cldr/unittest/web/TestMisc.java

```diff
@@ -82,27 +83,26 @@ public void TestIsCoverageOrganization() {
     }
 
     public void TestGitHash() {
-        final String appsVersion = CLDRConfigImpl.getGitHashForSlug("CLDR-Apps");
-        assertNotNull("getting CLDR-Apps version", appsVersion);
-        /*
-         * TODO: fail if CLDRURLS.UNKNOWN_REVISION.equals(appsVersion))
-         * and likewise for toolsVersion
-         */
-        // if (CLDRURLS.UNKNOWN_REVISION.equals(appsVersion)) {
-        //    errln("❌ appsVersion = UNKNOWN_REVISION: " + appsVersion);
-        // }
-
-        final String toolsVersion = CLDRConfigImpl.getGitHashForSlug("CLDR-Tools");
-        assertNotNull("getting CLDR-Tools version", toolsVersion);
-
-        /*
-         * TODO: add a regression test here for CLDR_DATA_HASH
-         * Reference: https://unicode-org.atlassian.net/browse/CLDR-13582
-         */
-        // final String hash = CLDRConfig.getInstance().getProperty("CLDR_DATA_HASH");
-        // assertNotNull("getting CLDR_DATA_HASH", hash);
-        // if (hash != null && !hash.matches("[0-9a-f]+")) {
-        //    errln("❌ CLDR_DATA_HASH is not hex: " + hash);
-        // }
+        final String hash = CldrUtility.getCldrBaseDirHash();
+        assertNotNull("getCldrBaseDirHash", hash);
+        if (hash != null) {
+            if (!hash.matches("[0-9a-f]+")) {
+                errln("❌ getCldrBaseDirHash is not hex: " + hash);
```

Comment:
it could be 'unknown' if git is missing or if the code was compiled from a source tarball?maybe this should be a warning?

Reply:
That's a good point. On the other hand, if anyone runs Survey Tool in a context where the hash is unknown, cache busting will be broken. Warnings aren't always noticed in a timely fashion. It seems more important to prevent a regression in our actual usage, than to accommodate a hypothetical scenario.